course-content/34_machine_learning.qmd

```diff
@@ -0,0 +1,244 @@
+
+```{python}
+tips_df['big_tipper'] = tips_df['tip'].apply(lambda x: 0 if x < 2.9 else 1)
+tips_df.sample(5)
+```
+
+Well that seemed to work fine! The machine learning classification model that we will use can only use numeric data. You probably know from statistics that you can convert categorical variables into numeric ones by turning each of the subsets into discrete columns consisting of 1s and 0s. They are widely referred to as dummy variables. Dummies are like flags in the data, where 1 means a certain subset is "observed" and 0 means that particular subset is "absent". 
+
+Fortunately, there is a method called `get_dummies()` in pandas that can transform all non-numeric variables easily. In the code below, we pass in the whole dataframe and it will do its thing.
+
+```{python}
+tips_df = pd.get_dummies(tips_df)
+tips_df.head()
+```
+<br>
+At this point, we could go ahead and eliminate some of these newly created columns. For example, we could drop "time_Lunch" as this is essentially computationally redundant to "time_Dinner", just the exact obverse (i.e. if you know whether a meal was lunch or not, you know whether it was dinner or not) But we'll leave that for now, their continued presence in the data will not harm the model.
+
+## Identifying Target Variable and Features
+
+The next step is to define which variables are to be considered the independent variables (or features, in machine learning terms) and which singular dependent variable is the target variable that we want to predict. Otherwise, the program won't know what variables are used to predict what. 
+
+Independent variables are usually best thought of as those which are known at the point of decision. In our use case, we want to predict whether a party is likely to be a big tipper when they arrive at the restauarant.  At that point we likely know the number of people in the party, but not the total bill amount. So let's identify the independent variables or features by dropping the variables 'tip', 'total_bill', and 'big_tipper' from an object we will call "X". The dependent or target variable is easier to identify. That is simply "big_tipper". We will assign that a "y". Both "X" (typed usually as upper case X) and "y" (typed usually as lower case y) are widely used conventions in Python for these concepts. 
+
+Let's use what we learned about the `drop()` method to create the features object and assign to y the single column containing the target "big_tipper" variable.
+
+```{python}
+X = tips_df.drop(['tip', 'total_bill', 'big_tipper'], axis=1)
+y = tips_df['big_tipper']
+```
+
+One of the key requirements of supervised machine learning methods is to split the dataset up into training and test samples. For the training set, the "machine" will "learn" which restaurant parties are more likely to be big tippers and use that knowledge to create an algorithm that can be applied to the test set. 
```

Comment:
I'd probably include a reason, or a link to an article, as to why we want training/test samples. I assume that none of this will be common knowledge for anyone.

Reply:
Fair!